dftimewolf/lib/processors/turbinia_gcp_tam.py

```diff
@@ -0,0 +1,361 @@
+    self.turbinia_zone = turbinia_zone
+    self.sketch_id = sketch_id
+    self.run_all_jobs = run_all_jobs
+
+    turbinia_config.LoadConfig(config_file=self.turbinia_config_file)
+    if not self.project:
+      self.project = turbinia_config.TURBINIA_PROJECT
+    if not self.turbinia_zone:
+      self.turbinia_zone = turbinia_config.TURBINIA_ZONE
+
+    if self.project is None or self.turbinia_zone is None:
+      self.ModuleError(
+          'project or turbinia_zone are not all specified, bailing out',
+          critical=True)
+      return
+
+    self.turbinia_region = turbinia_config.TURBINIA_REGION
+    self.instance = turbinia_config.INSTANCE_ID
+    if turbinia_config.TURBINIA_PROJECT != self.project:
+      self.ModuleError(
+          'Specified project {0!s} does not match Turbinia configured '
+          'project {1!s}. Use gcp_turbinia_disk_copy_ts recipe to copy the '
+          'disk into the same project.'.format(
+              self.project, turbinia_config.TURBINIA_PROJECT), critical=True)
+      return
+    self._output_path = tempfile.mkdtemp()
+    self.client = turbinia_client.get_turbinia_client(run_local=False)
+
+  def TurbiniaProcess(
+      self, evidence_: evidence.Evidence) -> List[Dict[str, Any]]:
+    """Creates and sends a Turbinia processing request.
+
+    Args:
+      evidence_(turbinia.evidence.Evidence): The evience to process
+
+    Returns:
+      list[dict]: The Turbinia task data
+    """
+    try:
+      evidence_.validate()
+    except TurbiniaException as exception:
+      self.ModuleError(str(exception), critical=True)
+
+    request = TurbiniaRequest(requester=getpass.getuser())
+    request.evidence.append(evidence_)
+    if self.sketch_id:
+      request.recipe['globals']['sketch_id'] = self.sketch_id
+    if not self.run_all_jobs:
+      request.recipe['globals']['jobs_denylist'] = [
+          'StringsJob', 'BinaryExtractorJob', 'BulkExtractorJob', 'PhotorecJob']
+
+    threatintel = self.state.GetContainers(containers.ThreatIntelligence)
+    if threatintel:
+      self.logger.info(
+          'Sending {0:d} threatintel to Turbinia GrepWorkers...'.format(
+              len(threatintel)))
+      indicators = [item.indicator for item in threatintel]
+      request.recipe['globals']['filter_patterns'] = indicators
+
+    request_dict = {
+        'instance': self.instance,
+        'project': self.project,
+        'region': self.turbinia_region,
+        'request_id': request.request_id
+    }
+
+    task_data = []  # type: List[Dict[str, str]]
+    try:
+      self.logger.success(
+          'Creating Turbinia request {0:s} with Evidence {1!s}'.format(
+              request.request_id, evidence_.name))
+      self.client.send_request(request)
+      self.logger.info('Waiting for Turbinia request {0:s} to complete'.format(
+          request.request_id))
+      self.client.wait_for_request(**request_dict)
+      task_data = self.client.get_task_data(**request_dict)
```

Comment:
Can we move the logger statements outside the try block?

Reply:
We've got a logging statement either side of `send_request` so would require 2 try/except blocks, so I'd prefer not to fo it looks a bit cleaner.